text/0005-architecture-reviews.md

```diff
@@ -0,0 +1,163 @@
+    - scenarios that exercise the relevant forces against the proposed architecture
+    - how the proposal handles these scenarios
+  * A competitive analysis suggesting alternatives, costs, and opportunities
+
+3. The chair creates a list of questions to be discussed during the review. These questions should be:
+  * Open-ended
+  * Made available to the the team prior to the review so they can prepare answers
+
+4. The chair convenes a review meeting. The discussions of this meeting should be carefully minuted. The meeting comprises:
+  * The team making the proposal
+  * The chair
+  * Other senior engineers who will provide valuable input
+
+The review meeting should be a discussion of the issues, but should avoid feeling pressured to make a decision. The goal is to understand the issues raised by the question(s) and the background from the review packet, and to add to it wisdom from the people at the review. The meeting may decide to alter the questions and undergo a subsequent review. We should be very open about conversations that happen after the review meeting.
+
+The ideal review process scales well. The same basic system should work for a quick 2 person decision over the best way to design a certain feature, and for a critical organization-wide decision about a path to take. The term “team” is used above although we strongly recommend design reviews for smaller questions. If you feel yourself wondering if some design is best, it should be easy to perform a review.
+
+For both a Roadmap Review and a Design Review the goal is to hear perspectives that will lead to a better definition of a problem or solution, and better judgment about whether it should be solved or how to solve it.
+
+## Rationale
+
+### Key considerations
+
+* What are the desired **outputs** of the review process?
+* Which **audience** is intended to consume the outputs of the review process?
+* What committee **structure** is best positioned to deliver those outputs?
+* What are the **inputs** that the committee should evaluate?
+* What **process** will most encourage participants to look at review as an asset rather than an obstacle?
+* What kinds of **conduct** are necessary in order for the process to be inclusive and constructive?
+* What size and **maturity** should projects achieve before undergoing review?
+* When should review results be **disseminated**?
+
+### Outputs and Audience
+
+The first two considerations define the goals of the review process and are two sides of the same coin.
+
+The audience of a Roadmap Review is those "up the chain" of decision makers and the audience of a Design Review is those running “down the chain” of implementers.
+
+A Roadmap Review should focus on:
+
+* The problem space
+* Large forces impacting Mozilla’s choices in the space
+* Competitive analysis
+* Opportunity cost
+
+The key outcome is senior management investing in the problem space and building consensus around a shared reality.
```

Comment:
I think the other key outcome is providing some indication of how success of the project can be evaluated down the line, as well as documentation of what the fundamental assumptions (e.g. about market structure) the project is based on are.

The latter is especially important if it turns out, as the project progresses, that some of those fundamental assumptions are incorrect, or become incorrect.  Understanding what they are should provide us built-in triggers that might trigger a re-evaluation of the roadmap.

Reply:
Agreed.

> The key outcome is firstly, senior management investing in the problem space and building consensus around a shared reality. Secondly, to provide documentation of both the goals and assumptions of the project to aid later evaluation.